File: analytics/src/main/java/io/appmetrica/analytics/impl/preparer/BytesValueComposer.java
package io.appmetrica.analytics.impl.preparer;

import android.util.Base64;
import androidx.annotation.NonNull;
import io.appmetrica.analytics.impl.request.ReportRequestConfig;
import io.appmetrica.analytics.logger.appmetrica.internal.DebugLogger;

public class BytesValueComposer implements ValueComposer {

    private static final String TAG = "[BytesValueComposer]";

    @NonNull
    @Override
    public byte[] getValue(@NonNull EventFromDbModel event, @NonNull ReportRequestConfig config) {
        byte[] value = new byte[0];
        if (event.getValue() != null) {
            try {
                value = Base64.decode(event.getValue(), Base64.DEFAULT);
            } catch (Throwable e) {
                DebugLogger.INSTANCE.error(
                    TAG,
                    e,
                    "Something went wrong while decoding base 64 event value."
                );
            }
        }
        return value;
    }
}

File: analytics/src/main/java/io/appmetrica/analytics/impl/preparer/BytesValueComposer.kt
package io.appmetrica.analytics.impl.preparer

import android.util.Base64
import io.appmetrica.analytics.impl.request.ReportRequestConfig
import io.appmetrica.analytics.logger.appmetrica.internal.DebugLogger

internal class BytesValueComposer : ValueComposer {

    private val tag = "[BytesValueComposer]"

    override fun getValue(event: EventFromDbModel, config: ReportRequestConfig): ByteArray {
        return event.foldValue(
            onLegacy = { legacy ->
                try {
                    Base64.decode(legacy, Base64.DEFAULT)
                } catch (e: Throwable) {
                    DebugLogger.error(
                        tag,
                        e,
                        "Something went wrong while decoding base 64 event value."
                    )
                    ByteArray(0)
                }
            },
            onBytes = { it },
            onEmpty = { ByteArray(0) },
        )
    }
}
